### PR TITLE
SearchKit - Allow super admins to disable Search Display access checks

### PR DESCRIPTION
**Pull request by colemanw**
Overview
----------------------------------------
This allows users with 'all CiviCRM permissions and ACLs' to configure a search display to bypass permission checks and display all records to the user.
image

Technical Details
----------------------------------------
Once a display is set to bypass ACLs, it can only be edited by a super-admin, ordinary admin users will not be able to edit the display or the saved search.

Such a display will not automatically appear on its own page; it must be embedded in an Afform, and the Afform will act as gatekeeper for users to view the display.

Comments
---------
This builds upon the groundwork done in #19797 and #20533

### CODE REVIEW

File: ext/search_kit/CRM/Search/Upgrader.php [hunk 1]
```diff
@@ -139,9 +139,39 @@ public function upgrade_1003() {
...
+    $this->ctx->log->info('Applying update 1004 - fix menu permission.');
     CRM_Core_DAO::executeQuery("UPDATE civicrm_navigation SET permission = 'administer CiviCRM data' WHERE url = 'civicrm/admin/search'");
     return TRUE;
   }
 
+  /**
+   * Upgrade 1005 - add acl_bypass column.
+   * @return bool
+   */
+  public function upgrade_1005() {
```
Comment (seamuslee001) [inline ext/search_kit/CRM/Search/Upgrader.php:151]: So this feels right but just making a point that these Extension upgrades are only run via the Extension Upgrade process not the main upgrade process but it should flag that they need to be done anyway.

**Comment by seamuslee001** [comment]
So @colemanw one question on this is that does this still provide any field level permission handling or does that get completely by-passed as well?

**Comment by seamuslee001** [comment]
@colemanw I tested this on the PR test site and it seems like it works well but want to do more r-run testing and one thing I just considered is that if a search display cannot be used at all if permissions are disabled without embedding on an afform, should the ability to disable permission only show if afform is enabled?

**Comment by colemanw** [comment]
> So @colemanw one question on this is that does this still provide any field level permission handling or does that get completely by-passed as well?

All permissions will be by-passed, BUT the search display will never show fields that are not configured to be part of the search, and it will never allow filters that are not explicitly part of the select clause or a filter field on the afform.

**Comment by colemanw** [comment]
> should the ability to disable permission only show if afform is enabled?

Well, maybe. I didn't want to rule out the possibility of a developer embedding a search display on some other (non-afform) Angular screen tho.

File: ext/search_kit/CRM/Search/Upgrader.php (hunk 1 above)
Comment (colemanw) [reply ext/search_kit/CRM/Search/Upgrader.php:151]: IMO we ought to run extension upgrades as part of the main upgrader.

**Comment by seamuslee001** [comment]
I think my concerns are satisfied here and it worked in my testing merging.

**Pull request merged**